featuretools/entityset/entityset.py

```diff
@@ -246,265 +260,281 @@ def add_relationships(self, relationships):
 
-    def entity_from_dataframe(self,
-                              entity_id,
-                              dataframe,
-                              index=None,
-                              variable_types=None,
-                              make_index=False,
-                              time_index=None,
-                              secondary_time_index=None,
-                              already_sorted=False):
+    def add_dataframe(self,
+                      dataframe_name,
+                      dataframe,
+                      index=None,
+                      logical_types=None,
+                      semantic_tags=None,
+                      make_index=False,
+                      time_index=None,
+                      secondary_time_index=None,
+                      already_sorted=False):
         """
-        Load the data for a specified entity from a Pandas DataFrame.
+        Add a DataFrame to the EntitySet with Woodwork typing information.
 
         Args:
-            entity_id (str) : Unique id to associate with this entity.
+            dataframe_name (str) : Unique name to associate with this dataframe.
 
             dataframe (pandas.DataFrame) : Dataframe containing the data.
 
-            index (str, optional): Name of the variable used to index the entity.
-                If None, take the first column.
+            index (str, optional): Name of the column used to index the dataframe.
+                Must be unique. If None, take the first column.
+
+            logical_types (dict[str -> Woodwork.LogicalTypes/str, optional]):
+                Keys are column names and values are logical types. Will be inferred if not specified.
 
-            variable_types (dict[str -> Variable/str], optional):
-                Keys are of variable ids and values are variable types or type_strings. Used to to
-                initialize an entity's store.
+            semantic_tags (dict[str -> str/set], optional):
+                Keys are column names and values are semantic tags.
 
             make_index (bool, optional) : If True, assume index does not
                 exist as a column in dataframe, and create a new column of that name
                 using integers. Otherwise, assume index exists.
 
-            time_index (str, optional): Name of the variable containing
-                time data. Type must be in :class:`variables.DateTime` or be
-                able to be cast to datetime (e.g. str, float, or numeric.)
+            time_index (str, optional): Name of the column containing
+                time data. Type must be numeric or datetime in nature.
 
-            secondary_time_index (dict[str -> Variable]): Name of variable
-                containing time data to use a second time index for the entity.
+            secondary_time_index (dict[str -> list[str]]): Name of column
+                containing time data to use a second time index for the dataframe.
```

Comment:
That seems like a good idea! 
> The dictionary key is the column in the dataframe to be used as the secondary time index, and the value is the list of columns associated with that secondary time index.

This has gotten me thinking: do we allow multiple secondary time indexes to be set? Because in `normalize_dataframe` we specify for `make_secondary_time_index` that "only one secondary time index is allowed" but I don't think we actually restrict this anywhere.